packages/backend-core/src/features/index.ts

```diff
@@ -1,79 +1,108 @@
 import env from "../environment"
 import * as context from "../context"
+import { cloneDeep } from "lodash"
 
-export * from "./installation"
-
-/**
- * Read the TENANT_FEATURE_FLAGS env var and return an array of features flags for each tenant.
- * The env var is formatted as:
- *  tenant1:feature1:feature2,tenant2:feature1
- */
-export function buildFeatureFlags() {
-  if (!env.TENANT_FEATURE_FLAGS) {
-    return
+class Flag<T> {
+  static withDefault<T>(value: T) {
+    return new Flag(value)
   }
 
-  const tenantFeatureFlags: Record<string, string[]> = {}
+  private constructor(public defaultValue: T) {}
+}
 
-  env.TENANT_FEATURE_FLAGS.split(",").forEach(tenantToFeatures => {
-    const [tenantId, ...features] = tenantToFeatures.split(":")
+// This is the primary source of truth for feature flags. If you want to add a
+// new flag, add it here and use the `fetch` and `get` functions to access it.
+// All of the machinery in this file is to make sure that flags have their
+// default values set correctly and their types flow through the system.
+const FLAGS = {
+  LICENSING: Flag.withDefault(false),
+  GOOGLE_SHEETS: Flag.withDefault(false),
+  USER_GROUPS: Flag.withDefault(false),
+  ONBOARDING_TOUR: Flag.withDefault(false),
+}
 
-    features.forEach(feature => {
-      if (!tenantFeatureFlags[tenantId]) {
-        tenantFeatureFlags[tenantId] = []
-      }
-      tenantFeatureFlags[tenantId].push(feature)
-    })
-  })
+const DEFAULTS = Object.keys(FLAGS).reduce((acc, key) => {
+  const typedKey = key as keyof typeof FLAGS
+  // @ts-ignore
+  acc[typedKey] = FLAGS[typedKey].defaultValue
```

Comment:
I couldn't find a nice way to make this type check because `FLAGS[typedKey].defaultValue` will have a type that's the union of all flags. So it's currently a boolean but when you add, say, a number flag it'll fail. I suspect it's possible to make this check correctly but I couldn't figure it out.